kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -1176,6 +1218,19 @@ public void testRecordKeysAndHeadersWritten() throws Exception {
     );
   }
 
+  private RecordWriterProvider<S3SinkConnectorConfig> getKeyHeaderValueProviderJsonHeaders() {
+    // setup header record provider for writing record header files.
+    RecordWriterProvider<S3SinkConnectorConfig> headerWriterProvider =
+        new JsonFormat(storage).getRecordWriterProvider();
+    ((RecordViewSetter) headerWriterProvider).setRecordView(new HeaderRecordView());
+    // initialize the KVHWriterProvider with header and key writers turned on.
+    return new KeyValueHeaderRecordWriterProvider(
+        new AvroFormat(storage).getRecordWriterProvider(),
```

Comment:
why is this avro?

Reply:
The first parameter for the `KeyValueHeaderRecordWriterProvider()` refers to the converter for the record value, for the tests in this PR the main focus is the converter for the headers to be json, the first converter could be any of the formats available.